UOP1_Project/Assets/Scripts/Pool/Example/ParticlePool.cs

```diff
@@ -0,0 +1,38 @@
+﻿using UnityEngine;
+using OP1.Pool;
+using OP1.Factory;
+
+[CreateAssetMenu(fileName = "New Particle Pool", menuName = "Pool/Particle Pool")]
+public class ParticlePool : ComponentPool<PoolableParticle>
+{
+	[SerializeField]
+	private ParticleFactory _factory;
+	[SerializeField]
+	private int _initialPoolSize;
+
+	public override IFactory<PoolableParticle> Factory
+	{
+		get
+		{
+			return _factory;
+		}
+		set
+		{
+			_factory = value as ParticleFactory;
+		}
```

Comment:
This breaks CreateInstance. Though I'd like if you could expand on this: "add the setter to the derived class, but there's no need to enforce it from the base." If we can accomplish that without breaking CreateInstance, I'd be all for it.

Reply:
Well if you have a base class with a getter:
``` csharp
public abstract class BaseClass
{
    public abstract float Value { get; }
}
```
You can still add a setter to Value in any class that derives from it:
``` csharp
public class DerivedClass : BaseClass
{
    public float Value { get; set; }
}
```
So if you need a Factory setter for a specific pool, you can make that specific pool have it, and you'd be able to use CreateInstance on it.

Reply:
Hmm, that doesn't look right. In that code, you're shadowing (sans new keyword). And even if you do that, you still need to implement the proper override. And a proper override would error that the abstract `Value` does not have an overridable set accessor.

I might be missing something. Do you have this working? If so, please submit a pull and I'll take a look.

Reply:
Huh, sorry, my bad, you're totally right! I mixed everything up with interfaces, I was so certain I didn't even check 🤦🏻.
``` csharp
public interface IInterface
{
	float Value { get; }
}
public class MyClass : IInterface
{
	public float Value { get; set; }
}
```

Reply:
Ah, gotcha. No worries! I still don't think the factory belongs in the interface, but I'll leave this conversation open in case other reviewers would like to offer their opinion.